frontend/.storybook/decorators/with-theme.ts

```diff
@@ -0,0 +1,79 @@
+import { watch, onMounted, reactive, h } from "vue"
+import { useEffect } from "@storybook/preview-api"
+
+import { EffectiveColorMode } from "~/types/ui"
+
+import { useDarkMode } from "~/composables/use-dark-mode"
+import { useUiStore } from "~/stores/ui"
+
+import VThemeSelect from "~/components/VThemeSelect/VThemeSelect.vue"
+
+const themeState = reactive<{ value: EffectiveColorMode }>({ value: "light" })
+
+const setElementTheme = (
+  el: HTMLElement,
+  cssClass: `${EffectiveColorMode}-mode`
+) => {
+  if (cssClass === "dark-mode") {
+    el.classList.add("dark-mode")
+    el.classList.remove("light-mode")
+  } else {
+    el.classList.add("light-mode")
+    el.classList.remove("dark-mode")
+  }
+}
+
+/**
+ * Decorator to add the Storybook theme switcher to the addon toolbar, and the Openverse
+ * theme switcher to the bottom of the screen.
+ * We cannot use the toolbar during the tests that open an iframe without the toolbars,
+ * so we need to add the theme switcher to the bottom of the screen.
+ * The state of both is kept in sync.
+ */
+export const WithTheme = (story, context) => {
+  useEffect(() => {
+    themeState.value = context.globals.theme
+  }, [context.globals.theme])
+
+  return {
+    components: { story },
+    setup() {
+      const { cssClass } = useDarkMode()
+      const uiStore = useUiStore()
+
+      onMounted(() => {
+        document.body.classList.add("bg-default")
+
+        watch(
```

Comment:
(Just curiousity) What is the reason to have these `watch` calls inside `onMounted`? Does this effectively delay starting the watching until after the component is ready?

Reply:
I just used the same structure as the `withRtl` decorator. After your comment, I went back and refactored this. Moved the `watch` calls out of `onMounted`. I also found a way of using `useGlobals` (it didn't work when I try running it in the `onMounted`, but worked when I moved it out of it), so now the global theme is updated correctly when the Openverse theme switcher value is changed.

Reply:
I'll make an issue to also refactor the `withRtl` decorator so that when `dir` is changed using Openverse footer, the global `languageDirection` is updated, similar to how `with-theme` works now.